contracts/modules/v2/Assessment.sol

```diff
@@ -0,0 +1,276 @@
+    return votesOf[assessor].length;
+  }
+
+  function getClaimsCount() external override view returns (uint) {
+    return claims.length;
+  }
+
+  function getIncidentsCount() external override view returns (uint) {
+    return incidents.length;
+  }
+
+  /* === MUTATIVE FUNCTIONS ==== */
+
+  /**
+   *  Submits a claim for assessment
+   *
+   *  @dev This function requires an ETH submission fee. See: _getSubmissionFee()
+   *
+   *  @param coverId          Cover identifier
+   *  @param requestedAmount  The amount expected to be received at payout
+   *  @param withProof        When true, a ProofSubmitted event is emitted with ipfsProofHash.
+   *                          When false, no ProofSubmitted event is emitted to save gas if the
+   *                          cover wording doesn't enforce a proof of loss.
+   *  @param ipfsProofHash    The IPFS hash required for proof of loss. It is ignored if withProof
+   *                          is false
+   */
+  function submitClaim(
+    uint24 coverId,
+    uint96 requestedAmount,
+    bool withProof,
+    string calldata ipfsProofHash
+  ) external payable override onlyMember {
+    AssessmentClaimsLib.submitClaim(
+      config,
+      internalContracts,
+      claims,
+      claimants,
+      coverId,
+      requestedAmount,
+      withProof,
+      ipfsProofHash
+    );
+
+  }
+
+  function submitIncident(
+    uint24 productId,
+    uint96 priceBefore,
+    uint32 date
+  ) external override {
+    (
+      AffectedToken memory affectedToken,
+      Incident memory incident
+    ) = AssessmentIncidentsLib.getIncidentToSubmit(
+      config,
+      memberRoles(),
+      productId,
+      priceBefore,
+      date
+    );
+
+    AssessmentIncidentsLib.saveIncident (
+      incident,
+      incidents,
+      affectedToken,
+      tokenAffectedByIncident
+    );
+  }
+
+  function depositStake (uint96 amount) external override onlyMember {
+    Stake storage stake = stakeOf[msg.sender];
+    stake.amount += amount;
+    nxm.transferFrom(msg.sender, address(this), amount);
+  }
+
+  function withdrawReward (address user, uint104 untilIndex) external override {
+    AssessmentVoteLib.withdrawReward(
+      config,
+      nxm,
+      user,
+      untilIndex,
+      stakeOf,
+      votesOf,
+      claims,
+      incidents
+    );
+  }
+
+  function withdrawStake (uint96 amount) external override onlyMember {
+    AssessmentVoteLib.withdrawStake(config, nxm, stakeOf, votesOf, amount);
+  }
+
+  function redeemClaimPayout (uint104 id) external override {
+    AssessmentClaimsLib.redeemClaimPayout(
+      config,
+      internalContracts,
+      claims,
+      claimants,
+      id
+    );
+  }
+
+  function redeemIncidentPayout (uint104 incidentId, uint32 coverId, uint depeggedTokens)
+  external override {
+    AssessmentIncidentsLib.redeemIncidentPayout(
+      internalContracts,
+      incidents[incidentId],
```

Comment:
does everything work as expected when the incidentId doesn't exist? should we explictly revert?

Reply:
It implicitly reverts if the index of an array >= length.